### PR TITLE
IN-14-http-handling

### PR DESCRIPTION
**Pull request by SamanNsr**
# Added:
- echo http server 
- http router and controller
- http client
# Fixed:
- remove api error
- fix external error handling

### CODE REVIEW

File: envs/local.env [hunk 1]
```diff
@@ -5,6 +5,13 @@ APP_ENV = prod
 GRPC_HOST=localhost
 GRPC_PORT=3000
 
+# HTTP
+HTTP_PORT=4000
+
+# GRPC clients
+GO_TEMPLATE_GRPC_PORT=3000
```
Comment (AmiralaDev) [inline envs/local.env]: Let's just have `HTTP_PORT` and `GRPC_PORT`.
It does not related to project name.

File: envs/production.env [hunk 2]
```diff
@@ -5,6 +5,13 @@ APP_ENV = prod
 GRPC_HOST=localhost
 GRPC_PORT=3000
 
+# HTTP
+HTTP_PORT=4000
+
+# GRPC clients
+GO_TEMPLATE_GRPC_PORT=3000
+GO_TEMPLATE_GRPC_HOST=localhost
```
Comment (AmiralaDev) [inline envs/production.env]: Also Here *

File: envs/stage.env [hunk 3]
```diff
@@ -5,6 +5,13 @@ APP_ENV = prod
 GRPC_HOST=localhost
 GRPC_PORT=3000
 
+# HTTP
+HTTP_PORT=4000
+
+# GRPC clients
+GO_TEMPLATE_GRPC_PORT=3000
+GO_TEMPLATE_GRPC_HOST=localhost
```
Comment (AmiralaDev) [inline envs/stage.env]: Also Here *

File: envs/test.env [hunk 4]
```diff
@@ -5,6 +5,13 @@ APP_ENV = prod
 GRPC_HOST=localhost
 GRPC_PORT=3000
 
+# HTTP
+HTTP_PORT=4000
+
+# GRPC clients
+GO_TEMPLATE_GRPC_PORT=3000
+GO_TEMPLATE_GRPC_HOST=localhost
```
Comment (AmiralaDev) [inline envs/test.env]: Also Here *

File: envs/local.env (hunk 1 above)
Comment (SamanNsr) [reply envs/local.env]: Sure, but this is for the external client that has been used in external layer.

File: pkg/constant/constant.go [hunk 5]
```diff
@@ -22,3 +25,16 @@ const (
...
+	MaxHeaderBytes       = 1 << 20
+	StackSize            = 1 << 10 // 1 KB
+	BodyLimit            = "2M"
+	ReadTimeout          = 15 * time.Second
+	WriteTimeout         = 15 * time.Second
+	GzipLevel            = 5
+	WaitShotDownDuration = 3 * time.Second
+	Dev                  = "development"
+	Test                 = "test"
+	Production           = "production"
```
Comment (AmiralaDev) [inline pkg/constant/constant.go]: Do we have any conflict with line 6-8 in this file ??
```
const (
	AppEnvProd = "prod"
	AppEnvDev  = "dev"
	AppEnvTest = "test"
)
```

**Review by AmiralaDev — APPROVED** [review]
Nice job Sam.

File: envs/local.env (hunk 1 above)
Comment (AmiralaDev) [reply envs/local.env]: @SamanNsr , So Its like `EXTERNAL_SERVICE_PORT` ?
 Thats Ok

Comment (SamanNsr) [reply envs/local.env]: Maybe `EXTERNAL_SERVICE_PORT` is better convention!

Comment (AmiralaDev) [reply envs/local.env]: @SamanNsr Thats ok for Template.
But Lets add `EXTERNAL` keyword here

File: pkg/constant/constant.go (hunk 5 above)
Comment (SamanNsr) [reply pkg/constant/constant.go]: I'll fix it.

File: envs/local.env (hunk 1 above)
Comment (SamanNsr) [reply envs/local.env]: Sure

**Pull request merged**